src/helper.cc

```diff
@@ -1286,13 +1298,10 @@ GetFirstAvailable(const helper * hlp)
     for (n = hlp->servers.head; n != NULL; n = n->next) {
         srv = (helper_server *)n->data;
 
-        if (srv->canAcceptRequests())
-            continue;
-
         if (selected && selected->stats.pending <= srv->stats.pending)
             continue;
 
-        if (srv->flags.shutdown)
+        if (!srv->available())
             continue;
 
         if (!srv->stats.pending)
```

Comment:
If this check is already covered by available(), please remove. Otherwise, please either add it to available() or discuss why it should not be added.